mlx/coverity_services.py

```diff
@@ -1,253 +1,203 @@
+
+    def retrieve_column_keys(self):
+        """Retrieves the column keys and associated display names.
+
+        Returns:
+            list[dict]: A list of dictionaries where the keys of each dictionary are 'columnKey' and 'name'
+        """
+        if not self._columns:
+            url = f"{self.base_url.rstrip('/')}/issues/columns?queryType=bySnapshot&retrieveGroupByColumns=false"
+            self._columns = self._request(url)
+        return self.columns
+
+    def retrieve_checkers(self):
+        """Retrieve the list of checkers from the server.
+
+        Returns:
+            list[str]: The list of valid checkers
+        """
         if not self.checkers:
-            self.checkers = self.client.service.getCheckerNames()
+            url = f"{self.base_url.rstrip('/')}/checkerAttributes/checker"
+            checkers = self._request(url)
+            if checkers and "checkerAttributedata" in checkers:
+                self._checkers = [checker["key"] for checker in checkers["checkerAttributedata"]]
         return self.checkers
 
+    def _request(self, url, data=None):
+        """Perform a POST or GET request to the specified url.
+        Uses a GET request when data is `None`, uses a POST request otherwise
+
+        Args:
+            url (str): The URL for the request
+            data (dict): Optional data to send
+
+        Returns:
+            dict: the content of server's response
+
+        Raises:
+            requests.HTTPError
+        """
+        if data:
+            response = self.session.post(url, json=data)
+        else:
+            response = self.session.get(url)
+        if response.ok:
+            return response.json()
+        try:
+            err_msg = response.json()["message"]
+        except (requests.exceptions.JSONDecodeError, KeyError):
+            err_msg = response.content.decode()
+        logger = getLogger("coverity_logging")
+        logger.warning(err_msg)
+        return response.raise_for_status()
+
     @staticmethod
-    def add_filter_rqt(name, req_csv, valid_list, filter_list, allow_regex=False):
-        '''Lookup the list of given filter possibility, add to filter spec and return a validated list'''
-        logging.info('Validate required %s [%s]', name, req_csv)
+    def add_filter_rqt(name, req_csv, valid_list, allow_regex=False):
+        """Add filter when the attribute is valid. If `valid_list` is not defined,
+        all attributes of the CSV list are valid.
+        The CSV list can allow regular expressions when `allow_regex` is set to True.
+
+        Args:
+            name (str): String representation of the attribute.
+            req_csv (str): A CSV list of attribute values to query.
+            valid_list (list/dict): The valid attributes.
+            allow_regex (bool, optional): True when regular expressions are allowed. Defaults to False.
+
+        Returns:
+            str: The validated CSV list
+            list[str]: The list of valid attributes
```

Comment:
The first return value is not really needed, so I would remove it. I would rename this function to `validate_filter_option`

Reply:
Resolved in 7591e84927588e99702ad5b0ff9b193149279178